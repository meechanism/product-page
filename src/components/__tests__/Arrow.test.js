import React from 'react';
import {
  ArrowLeft,
  ArrowRight,
  IconLeft,
  IconRight,
  ArrowContainer
} from '../Arrow';

describe('Arrow component', () => {
  const onClickFn = jest.fn();

  it('renders left arrow', () => {
    const output = shallow(<ArrowLeft onClick={onClickFn} />);
    expect(shallowToJson(output)).toMatchSnapshot();
    expect(output.find(IconLeft).exists()).toBeTruthy();

    const container = output.find(ArrowContainer);
    expect(container.exists()).toBeTruthy();
    expect(container.props().onClick).toBe(onClickFn);
  });
  it('renders right arrow', () => {
    const output = shallow(<ArrowRight onClick={onClickFn} />);
    expect(shallowToJson(output)).toMatchSnapshot();
    expect(output.find(IconRight).exists()).toBeTruthy();

    const container = output.find(ArrowContainer);
    expect(container.exists()).toBeTruthy();
    expect(container.props().onClick).toBe(onClickFn);
  });

  it('should trigger passed fn when left arrow is clicked', () => {
    const output = shallow(<ArrowLeft onClick={onClickFn} />);
    output.simulate('click');
    expect(onClickFn).toHaveBeenCalled();
  });

  it('should trigger passed fn when right arrow is clicked', () => {
    const output = shallow(<ArrowRight onClick={onClickFn} />);
    output.simulate('click');
    expect(onClickFn).toHaveBeenCalled();
  });
});
