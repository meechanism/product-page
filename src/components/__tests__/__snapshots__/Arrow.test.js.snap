// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow component renders left arrow 1`] = `
<styled.a
  href=""
  onClick={[MockFunction]}
>
  <styled.i />
</styled.a>
`;

exports[`Arrow component renders right arrow 1`] = `
<styled.a
  href=""
  onClick={[MockFunction]}
  right={true}
>
  <styled.i />
</styled.a>
`;
